### PR TITLE
Theme Hook Alliance

### PR DESCRIPTION
**Pull request by protechig**
I am working on an independent fork of _s to help my team (bourbon, neat, etc..). One thing that I love about the Genesis Framework is their robust hooks - which lead me to come across Doug Stewart's Theme Hook Alliance. I added them to my theme. I created a separate THA branch that only adds the THA functionality so it could be merged directly to _s if you choose. I saw an issue posted in January about it and the conclusion is that it would be considered. Here it is! This is my first github pull request.

Thanks, and Merry Christmas!

### CODE REVIEW

**Comment by jacklenox** [comment]
It's my understanding that the THA is primarily focussed on improving child theming. As @emiluzelac said above `_s` is a starter theme and creating a child of it is specifically discouraged.

I also don't see how these hooks are a good fit.

**Comment by grappler** [comment]
@jacklenox '_s' is a starter theme but all of the themes built off it will be parent themes so it makes sense for the parent theme based on _s to improve child themeing. These hooks can also be used in plugins. There is even a plugin on w.org which uses the hooks.

**Comment by jacklenox** [comment]
@grappler Good point. :)

**Comment by samikeijonen** [comment]
Is there a reason why hooks `tha_entry_before` and `tha_entry_top` are basically in the same spot in `404.php` file?

**Comment by ianstewart** [comment]
I'm not a fan of adding THA hooks to _s. It moves very far away from the leanness of _s and tons of hooks doesn't necessarily make child theming better.

**Comment by sterndata** [comment]
I've been loosely following this thead and I have to agree with ianstewart:  This isn't Genesis. It's lean, mean, quick.

File: content-search.php [hunk 1]
```diff
@@ -27,3 +28,4 @@
 		<?php _s_entry_footer(); ?>
 	</footer><!-- .entry-footer -->
 </article><!-- #post-## -->
+<?php tha_content_bottom()l ?>
```
Comment (grappler) [inline content-search.php]: There is an error here. The `1` should be `;`

**Comment by grappler** [comment]
This would be an interesting feature as discussed in https://github.com/Automattic/_s/issues/659 People are using `_s` for differnt thing e.g. Custom themes for clients, themes to be distributed to the mass or WordPress.com themes. Perhaps we could add this as an option which is disabled by default. It would make it easier for some of us if we could download `_s` with the hooks.

This is not just for child themes but also having standard hooks which plugins can hook into.

For refernce THA came out of this WP core ticket discussion: https://core.trac.wordpress.org/ticket/21506

File: content-search.php (hunk 1 above)
Comment (protechig) [inline content-search.php]: I'll get that fixed!

**Comment by protechig** [comment]
I'm impartial as to whether THA makes sense in `_s` or not. While `_s` isn't Genesis (obviously) one of the features that I really do enjoy about genesis is it's plethora of action hooks. That is why I forked it and added it to my fork - as I want to make a pseduo-framework / parent theme that suits my personal and professional needs. In the nature of GPL, I figured I'd submit a pull request and if enough people found it useful or wanted  then that's great!

I definitely do think `_s` does have multiple audiences as @grappler referenced, and it may be useful to the community to have different branches for different types of themes (.com vs. org) as well as be able to enable different features on download - why couldn't THA be a checkbox just like Sass is?

**Comment by obenland** [comment]
I agree with @emiluzelac, @jacklenox, and @ianstewart. It's just not a good fit at the moment, even as an optional feature.

Let's revisit this once there is a core supported solutions in place.

Thanks for your work on this PR @protechig!

**Pull request closed without merging**